sonyflake.go

```diff
@@ -274,3 +274,24 @@ func DecomposeToBuffer(id uint64, buf *Buffer) {
 	buf.MSB = uint8(msb)
 	buf.MachineID = uint16(machineID)
 }
+
+// Compose converts a Buffer into a sonyflake ID
+func (b Buffer) Compose() (uint64, error) {
+	if b.Time >= 1<<BitLenTime {
+		return 0, errors.New("over the time limit")
+	}
+
+	return uint64(b.Time)<<(BitLenSequence+BitLenMachineID) |
+		uint64(b.Sequence)<<BitLenMachineID |
+		uint64(b.MachineID), nil
+}
+
+// Inc increase the sequence number in the buffer by 1, or bumps to the next timestamp if that is not possible.
+func (b *Buffer) Inc() {
+	newSeq := b.Sequence + 1
+	if newSeq <= b.Sequence { // uint8 overflow
+		b.Time++
+		b.Sequence = 0
+	}
+	b.Sequence = newSeq
```

Comment:
Do we need the b.Sequence = 0, when we set it to the overflowed uint8 (Which is also 0) right after?

Reply:
good point, this is fishy... I think there is a missing return in the if branch. I'll add a test for this